plugins/hydra_joblib_launcher/hydra_plugins/hydra_joblib_launcher/joblib_launcher.py

```diff
@@ -0,0 +1,136 @@
+from hydra.types import TaskFunction
+
+log = logging.getLogger(__name__)
+
+
+class HydraJoblibLauncherSearchPathPlugin(SearchPathPlugin):
+    """
+    This plugin is allowing configuration files provided by the JoblibLauncher plugin to be discovered
+    and used once the JoblibLauncher plugin is installed
+    """
+
+    def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
+        # Appends the search path for this plugin to the end of the search path
+        search_path.append(
+            "hydra-joblib-launcher", "pkg://hydra_plugins.hydra_joblib_launcher.conf",
+        )
+
+
+class HydraJoblibLauncher(Launcher):
+    def __init__(self, joblib: Dict[str, Any] = {},) -> None:
+        """Joblib Launcher
+
+        Launches parallel jobs using Joblib.Parallel. For details, refer to:
+        https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html
+
+        This plugin is based on the idea and inital implementation of @emilemathieutmp:
+        https://github.com/facebookresearch/hydra/issues/357
+        """
+        self.config: Optional[DictConfig] = None
+        self.config_loader: Optional[ConfigLoader] = None
+        self.task_function: Optional[TaskFunction] = None
+
+        self.joblib = joblib
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+
+    def launch(self, job_overrides: Sequence[Sequence[str]]) -> Sequence[JobReturn]:
+        """
+        :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+        :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+        """
+        setup_globals()
+        assert self.config is not None
+        assert self.config_loader is not None
+        assert self.task_function is not None
+
+        configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
+        sweep_dir = Path(str(self.config.hydra.sweep.dir))
+        sweep_dir.mkdir(parents=True, exist_ok=True)
+        log.info(
+            "Joblib.Parallel({}) is launching {} jobs".format(
+                ",".join([f"{k}={v}" for k, v in self.joblib.items()]),
+                len(job_overrides),
+            )
+        )
+        log.info("Sweep output dir : {}".format(sweep_dir))
+
+        singleton_state = Singleton.get_state()
+
+        runs = Parallel(**self.joblib)(
+            delayed(dispatch_job)(
+                idx,
+                overrides,
+                self.config_loader,
+                self.config,
+                self.task_function,
+                singleton_state,
+            )
+            for idx, overrides in enumerate(job_overrides)
+        )
+
+        return runs  # type: ignore
```

Comment:
hmm, does the return type mismatches the expected one or is it just unknown?

Maybe you can assert the type for runs instead to tell mypy that the type is what you say it is (not sure how it work for containers though).

Reply:
The trouble here is that `runs = Parallel(delayed(...))`. These are both part of Joblib which does not provide type annotations. 

But we know that runs is `List[JobReturn]` in our case. I added manual assertions as you suggested and was able to remove the `# type: ignore`. 💯